.github/workflows/main.yml

```diff
@@ -359,6 +344,104 @@ jobs:
+        prepare-wave-hindcast-cache,
+        prepare-wind-hindcast-cache,
+      ]
+    if: |
+      always() &&
+      (
+        (
+          needs.prepare-nonhindcast-cache.result == 'success' &&
+          needs.prepare-wave-hindcast-cache.result == 'skipped' &&
+          needs.prepare-wind-hindcast-cache.result == 'skipped' &&
+          needs.check-changes.outputs.should-run-hindcast == 'false'    
+        ) ||
+        (
+          needs.prepare-nonhindcast-cache.result == 'success' &&
+          needs.prepare-wave-hindcast-cache.result == 'success' &&
+          needs.prepare-wind-hindcast-cache.result == 'success' &&
+          needs.check-changes.outputs.should-run-hindcast == 'true' 
+        )
+      )
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.11'
+
+      - name: Generate matrix
+        id: set-matrix
+        run: |
+          matrix_json=$(python .github/workflows/generate_notebook_matrix.py)
+          echo "$matrix_json" >> $GITHUB_OUTPUT
+
+  test-notebooks:
+    needs:
+      [
+        notebook-matrix,
+        check-changes,
+        prepare-nonhindcast-cache,
+        prepare-wave-hindcast-cache,
+        prepare-wind-hindcast-cache,
+      ]
+    strategy:
+      matrix: ${{ fromJson(needs.notebook-matrix.outputs.matrix) }}
+    if: |
+      always()
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up Miniconda
+        uses: conda-incubator/setup-miniconda@v3
+        with:
+          miniconda-version: 'latest'
+          auto-update-conda: true
+          python-version: '3.11'
+          activate-environment: TESTconda
+          use-only-tar-bz2: true
+
+      - name: Install dependencies
+        shell: bash -l {0}
+        run: |
+          conda install numpy cython pip hdf5 libnetcdf cftime netcdf4 --strict-channel-priority
+          pip install -e . --force-reinstall
+          python -m pip install --upgrade pip wheel
+          pip install nbval jupyter
+          pip install utm folium
+
+      - name: Ensure Conda environment is activated
+        shell: bash -l {0}
+        run: |
+          echo "source ~/miniconda3/etc/profile.d/conda.sh" >> ~/.bashrc
+          echo "conda activate TESTconda" >> ~/.bashrc
+          source ~/.bashrc
+
+      - name: Run notebook
+        shell: bash -l {0}
+        run: |
+          if [[ "${{ matrix.notebook }}" == "examples/metocean_example.ipynb" || "${{ matrix.notebook }}" == "examples/WPTO_hindcast_example.ipynb" ]]; then
+            if [[ "${{ needs.check-changes.outputs.should-run-hindcast }}" == 'true' ]]; then
+              jupyter nbconvert --to notebook --execute --inplace --ExecutePreprocessor.timeout=${{ matrix.timeout }} "${{ matrix.notebook }}"
```

Comment:
Yes each notebook in the repo has a custom timeout specified. It will fail if you go overtime. You can see some of these failures  on previous commits such as the commit before "Increase ADCP timeout".